lib/utils.cpp

```diff
@@ -39,7 +39,7 @@ findUncoveredRanges(const mem_access &access,
   return uncoveredRanges;
 }
 
-[[nodiscard]] OCSCache::Status simulateTrace(std::ifstream &trace, int n_lines,
```

Comment:
I think "sample n lines" could be renamed something more specific- maybe `sim_first_n_lines` or something? sample makes it seem like something more complex is involved